libdnf/base/goal.cpp

```diff
@@ -90,7 +88,13 @@ static const std::map<ProblemRules, const char *> PKG_PROBLEMS_DICT = {
     {ProblemRules::RULE_PKG_IMPLICIT_OBSOLETES, M_("package {0} implicitly obsoletes {1} provided by {2}")},
     {ProblemRules::RULE_PKG_REQUIRES, M_("package {1} requires {0}, but none of the providers can be installed")},
     {ProblemRules::RULE_PKG_SELF_CONFLICT, M_("package {1} conflicts with {0} provided by itself")},
-    {ProblemRules::RULE_YUMOBS, M_("both package {0} and {2} obsolete {1}")}};
+    {ProblemRules::RULE_YUMOBS, M_("both package {0} and {2} obsolete {1}")},
+    {ProblemRules::RULE_PKG_REMOVAL_OF_PROTECTED,
+     M_("The operation would result in removing"
+        " the following protected packages: {}")},
+    {ProblemRules::RULE_PKG_REMOVAL_OF_RUNNING_KERNEL,
+     M_("The operation would result in removing"
+        " of running kernel: {}")}};
 
 bool is_unique(
     const std::vector<std::pair<ProblemRules, std::vector<std::string>>> origin,
```

Comment:
Please pass vector by reference (do not copy).
```
const std::vector<std::pair<ProblemRules, std::vector<std::string>>> & origin,
```

Reply:
Fixed